src/tool/subcommands/api_cmd.rs

```diff
@@ -820,30 +839,33 @@ fn snapshot_tests(store: Arc<ManyCar>, n_tipsets: usize) -> anyhow::Result<Vec<R
                 tests.push(RpcTest::basic(
                     MpoolGetNonce::request((msg.from().into(),)).unwrap(),
                 ));
-                tests.push(RpcTest::identity(ApiInfo::state_list_messages_req(
+                tests.push(RpcTest::identity(StateListMessages::request((
                     MessageFilter {
-                        from: None,
+                        from: Some(msg.from()),
                         to: Some(msg.to()),
-                    },
-                    shared_tipset_key.into(),
-                    shared_tipset.epoch(),
-                )));
-                tests.push(RpcTest::identity(ApiInfo::state_list_messages_req(
+                    }
+                    .into(),
+                    LotusJson(tipset.key().into()),
+                    tipset.epoch().into(),
+                ))?));
+                tests.push(RpcTest::identity(StateListMessages::request((
                     MessageFilter {
-                        from: Some(msg.from()),
+                        from: Some(msg.to()),
                         to: None,
-                    },
-                    shared_tipset_key.into(),
-                    shared_tipset.epoch(),
-                )));
-                tests.push(RpcTest::identity(ApiInfo::state_list_messages_req(
+                    }
+                    .into(),
+                    LotusJson(tipset.key().into()),
+                    tipset.epoch().into(),
+                ))?));
+                tests.push(RpcTest::identity(StateListMessages::request((
                     MessageFilter {
                         from: None,
-                        to: None,
```

Comment:
Empty MessageFilter causes internal errors for both forest and lotus.  
If we want to keep this test case, we need to treat it a success when InternalErrors happen on both sides. @lemmih   @LesnyRumcajs what do you think?

Reply:
I'm okay with the condition that it's explicit for this particular test (like `should_fail`). I wouldn't like the behaviour like before, where all matching internal server errors were considered a success.

Reply:
Sounds great! Opened https://github.com/ChainSafe/forest/issues/4241 to track this feature.